src/ui/components/NavigationBar.js

```diff
@@ -0,0 +1,94 @@
+import { testStep } from "../../common/helpers/pwHelpers.js";
+
+export class NavigationBar {
+  constructor(page, userId = 0) {
+    this.page = page;
+    this.userId = userId;
+  }
+
+  step(title, fn) {
+    return testStep(title, fn, this.userId);
+  }
+
+  async clickNavLink(name) {
+    await this.page.getByRole("link", { name }).click();
+  }
+
+  async waitForPageUrl(urlPattern) {
+    await this.page.waitForURL(urlPattern, {
+      waitUntil: "domcontentloaded",
+    });
+  }
+
+  async waitForVisible(selector) {
+    await this.page.locator(selector).waitFor({ state: "visible" });
```

Comment:
```suggestion
    await this.page.locator(selector).waitFor();
```

`visible` is default param